### PR TITLE
Feature: transfer TLD

### PR DESCRIPTION
**Pull request by hbarcelos**
Implements [JP-799](https://janusproj.atlassian.net/secure/RapidBoard.jspa?rapidView=7&modal=detail&selectedIssue=JP-799) and [JP-803](https://janusproj.atlassian.net/secure/RapidBoard.jspa?rapidView=7&modal=detail&selectedIssue=JP-803):

- [x] Fixes ownership transfer for TLDs to update the reversed index (`owner -> TLD`).
     - [x] Added regression test case.
- [x] Implements ownership transfer for Domains.

### CODE REVIEW

File: packages/smartcontract/contracts/JanusNameService.sol [hunk 1]
```diff
@@ -26,21 +27,22 @@ contract JanusNameService {
         string storageHash;
         uint expires;
         address owner;
+        uint ownerIndex;
     }
 
-    TopDomain[] private topDomains;
```
Comment (rodrigorsdev) [inline packages/smartcontract/contracts/JanusNameService.sol:31]: why did you change the topDomains array and mappings to public?

File: packages/smartcontract/contracts/JanusNameService.sol [hunk 2]
```diff
@@ -26,21 +27,22 @@ contract JanusNameService {
...
-    mapping(bytes32 => uint) private topDomainsHashMap;
-    mapping(address => uint[]) private topDomainsOwnerMap;
+    TopDomain[] public topDomains;
+    mapping(bytes32 => uint) public topDomainsHashMap;
+    mapping(address => uint[]) public topDomainsOwnerMap;
 
-    Domain[] private domains;
```
Comment (rodrigorsdev) [inline packages/smartcontract/contracts/JanusNameService.sol:35]: why did you change the domains array and mappings to public?

File: packages/smartcontract/contracts/JanusNameService.sol (hunk 1 above)
Comment (hbarcelos) [reply packages/smartcontract/contracts/JanusNameService.sol:31]: It helped me debug the problem we had with updating the indexes for ownership transfer.
Furthermore, I added some regression tests that depend on this property being public.

File: packages/smartcontract/contracts/JanusNameService.sol (hunk 2 above)
Comment (hbarcelos) [reply packages/smartcontract/contracts/JanusNameService.sol:35]: Same reason as above.

File: packages/smartcontract/contracts/JanusNameService.sol (hunk 1 above)
Comment (rodrigorsdev) [reply packages/smartcontract/contracts/JanusNameService.sol:31]: yeah, i forgot about the tests.
Make sense.

File: packages/smartcontract/contracts/JanusNameService.sol (hunk 2 above)
Comment (rodrigorsdev) [reply packages/smartcontract/contracts/JanusNameService.sol:35]: yeah, i forgot about the tests.
Make sense.

File: packages/smartcontract/test/changeDomainOwnership.test.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      tld,
+      futureOwner,
+      { from: ownerAddress }
+    );
+
+    const [transfered, kept] = await Promise.all([
+      // The index 0 is already taken by the contract
+      contractInstance.domains(1),
+      contractInstance.domains(2),
+    ]);
```
Comment (hbarcelos) [inline packages/smartcontract/test/changeDomainOwnership.test.js:138]: @rodrigorsdev here's the regression test I was talking about.

**Review by rodrigorsdev — APPROVED** [review]
ok, nice job
